db/version_set.cc

```diff
@@ -6031,6 +6049,21 @@ Status VersionSet::WriteCurrentStateToManifest(
     if (!io_s.ok()) {
       return io_s;
     }
+
+    // WAL addition records above may contain entry for WAL
+    // that already has deletion record in the manifest.
+    // Such addition record should be offsetted by a deletion record.
+    VersionEdit wal_deletions;
+    wal_deletions.DeleteWalsBefore(min_log_number_to_keep());
```

Comment:
I'd group this with `SetMinLogNumberToKeep()` since it's essentially the same thing. Why we have such redundancy in the first place is an interesting question but I'm trying to ignore it for this PR.

Reply:
Fixed